helpers.go

```diff
@@ -162,54 +164,152 @@ func getCassandraBaseType(name string) Type {
 	}
 }
 
-func getCassandraType(name string, logger StdLogger) TypeInfo {
+// Parse long Java-style type definition to internal data structures.
+func getCassandraLongType(name string, protoVer byte, logger StdLogger) TypeInfo {
+	if strings.HasPrefix(name, SET_TYPE) {
+		return CollectionType{
+			NativeType: NewNativeType(protoVer, TypeSet),
+			Elem:       getCassandraLongType(strings.TrimPrefix(name[:len(name)-1], SET_TYPE+"("), protoVer, logger),
+		}
+	} else if strings.HasPrefix(name, LIST_TYPE) {
+		return CollectionType{
+			NativeType: NewNativeType(protoVer, TypeList),
+			Elem:       getCassandraLongType(strings.TrimPrefix(name[:len(name)-1], LIST_TYPE+"("), protoVer, logger),
+		}
+	} else if strings.HasPrefix(name, MAP_TYPE) {
+		names := splitJavaCompositeTypes(strings.TrimPrefix(name[:len(name)-1], MAP_TYPE+"("))
+		if len(names) != 2 {
+			logger.Printf("Error parsing map type, it has %d subelements, expecting 2\n", len(names))
```

Comment:
I think gocql log statements are usually prefixed by `gocql: ` if I'm not mistaken

Reply:
Yes, updated.